platforms.md

```diff
@@ -50,3 +50,57 @@ the `ore <platform> gc` command.
+ - If a security group matching the name in `aws-sg` (default: `kola`) is not found then one will be created, along with a VPC, internet gateway, route table, and subnets.
+ - Both AMI's as well as the channel tags are accepted via the `aws-ami` parameter, if a channel is given it will be resolved via the release bucket's `coreos_production_ami_all.json` file.
+
+## Azure
+
+ - The Azure platform wraps [azure-sdk-for-go](https://github.com/Azure/azure-sdk-for-go).
+ - For each cluster to be created a Resource Group is created containing the necessary networking resources and a storage account (which will be used to store boot diagnostics for any machines in the cluster).
+ - There are two types of images in Azure (published images and custom images). For using published images the channel can be passed via the `azure-sku` parameter and the version can be passed via the `azure-version` parameter. To specify a custom image you can pass the `azure-disk-uri` parameter.
+ - `kola` works entirely on ARM based authentication, `ore` has methods for both ASM or ARM credentials.
+ - `GC` in Azure searches for Resource Groups with a prefix of `kola-cluster` in the name.
+
+## DigitalOcean
+
+ - The DO platform wraps [godo](https://github.com/digitalocean/godo).
+ - DigitalOcean has no method for uploading custom images, as a result the `ore do create-image` command does a [~~terrifying~~ special workaround](https://github.com/coreos/mantle/blob/master/cmd/ore/do/create-image.go#L117-L173) which specifies custom userdata that does the following (after which the machine is snapshotted):
+   - configure networking in the initramfs
+   - Download a custom image
+   - replaces `/root/initramfs/shutdown` with a script to:
+      - unmount the root filesystem
+      - dd the custom image over the disk
+      - reboot
+ - SSH keys are passed via both DO metadata and the userdata.
+ - UserData is passed via the DO metadata service.
+ - Instances are given the tag `mantle` which is used by `GC`.
+ - The `do-image` parameter accepts snapshot ID's as well as the channel names, if a channel is given the relevant slug `coreos-<channel>` will be passed to DO.
+
+## ESX
+
+ - The ESX platform wraps [govmomi](https://github.com/vmware/govmomi).
+ - ESX is actually a bit of a misnomer, it actually requires that the target is a vSphere instance (as it uses features not present in the free ESXi hypervisor).
+ - The general workflow for running tests on ESX is to run the `ore esx create-base` to upload an OVF image and then that name is specified to `kola` via the `esx-base-vm` parameter. This is done to only require a clone operation rather than a full re-upload of the image.
```

Comment:
s/that name/the resulting image name/
s/only require/only require kola to perform/